File: GuiRunner/ValueConverters/TestStatusToFontWeightConverter.cs
using System;
using System.Globalization;
using System.Windows.Data;
using FeldSpar.Api.Engine.ClrInterop.ViewModels;
using FeldSparGuiCSharp.StyleConstants;

namespace FeldSparGuiCSharp.ValueConverters
{
    public class TestStatusToFontWeightConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            var status = (TestStatus) value;

            switch (status)
            {
                case TestStatus.Failure:
                    return TestStatusFontWeight.FailedFontWeight;
                case TestStatus.Ignored:
                    return TestStatusFontWeight.IgnoredFontWeight;
                case TestStatus.Running:
                    return TestStatusFontWeight.RunningFontWeight;
                case TestStatus.Success:
                    return TestStatusFontWeight.SuccessFontWeight;
                default:
                    return TestStatusFontWeight.NoneFontWeight;
            }
        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            throw new NotImplementedException();
        }
    }
}

File: FSharpWpf/TestStatusToFontWeightConverter.fs
namespace FeldSparGuiFSharp.StyleConstants
open System.Windows
open FeldSpar.Api.Engine.ClrInterop.ViewModels

module TestStatusFontWeight =
    let getFontWeight status = 
        match status with
        | TestStatus.Running | TestStatus.Failure | TestStatus.Ignored -> FontWeights.Bold
        | _ -> FontWeights.Normal

namespace FeldSparGuiFSharp.ValueConverters
open System
open System.Globalization
open System.Windows.Data
open FeldSpar.Api.Engine.ClrInterop.ViewModels
open FeldSparGuiFSharp.StyleConstants
open TestStatusFontWeight

type TestStatusToFontWeightConverter () = 
    interface IValueConverter with
        member this.Convert ((value:obj), (targetType:Type), (parameter:obj), (culture:CultureInfo)) =
            let status = value :?> TestStatus

            (getFontWeight status) :> obj

        member this.ConvertBack ((value:obj), (targetType:Type), (parameter:obj), (culture:CultureInfo)) =
            failwith "Not Implemented"

    member this.IValueConverter = this :> IValueConverter

    member this.Convert ((value:obj), (targetType:Type), (parameter:obj), (culture:CultureInfo)) = this.IValueConverter.Convert (value, targetType, parameter, culture)
    member this.ConvertBack ((value:obj), (targetType:Type), (parameter:obj), (culture:CultureInfo)) = this.IValueConverter.ConvertBack (value, targetType, parameter, culture)